### ISSUE TITLE
Unable to deserialize dataclass when used as task argument

### ISSUE THREAD

**Issue by ferntheplant** [samuelcolvin/arq#384]
In the following example I try to use a dataclass as a single argument to my worker function. However the worker gives a deserialization error because it can't find the `FooParams` dataclass definition (as seen by modifying `deserialize_job_raw` to log the original deserialization error). Running `main` with `params_as_dict` instead of as a dataclass works but then the worker function itself fails due to receiving a dict instead of a dataclass argument.

Is this a known bug? Are there any workarounds?

```python
import asyncio
from dataclasses import dataclass
from arq import create_pool
from arq.connections import RedisSettings

@dataclass
class FooParams:
    foo: str

async def foo(ctx: dict, params: FooParams) -> None:
    print(f"Found params {params.foo}")

class WorkerSettings:
    functions = [foo]

async def main():
    redis = await create_pool(RedisSettings())
    params_as_dict = { "foo": "hello world" }
    params_as_dc = FooParams(foo="world hello")
    await redis.enqueue_job('foo', params_as_dc)

if __name__ == '__main__':
    asyncio.run(main())
```

```
17:57:47: Starting worker for 1 functions: foo
17:57:47: redis_version=6.2.1 mem_usage=888.52K clients_connected=1 db_keys=13
17:57:50: deserializing job 8d9b5a66635e42f198e2371eb16461be failed
Traceback (most recent call last):
  File "/Users/fjorn/heyqq/hey-qq/peanutbutter/.venv/lib/python3.10/site-packages/arq/jobs.py", line 294, in deserialize_job_raw
    d = deserializer(r)
AttributeError: Can't get attribute 'FooParams' on <module '__main__' from '/Users/fjorn/heyqq/hey-qq/peanutbutter/.venv/bin/arq'>

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/Users/fjorn/heyqq/hey-qq/peanutbutter/.venv/lib/python3.10/site-packages/arq/worker.py", line 495, in run_job
    function_name, args, kwargs, enqueue_job_try, enqueue_time_ms = deserialize_job_raw(
  File "/Users/fjorn/heyqq/hey-qq/peanutbutter/.venv/lib/python3.10/site-packages/arq/jobs.py", line 297, in deserialize_job_raw
    raise DeserializationError('unable to deserialize job') from e
arq.jobs.DeserializationError: unable to deserialize job
```

**Comment by JonasKs** [comment]
Serialization and deserialization is done through `pickle`. See the docs [here](https://arq-docs.helpmanual.io/#custom-job-serializers). In other words, unless you can pickle a dataclass, you either [convert to dict](https://docs.python.org/3/library/dataclasses.html#dataclasses.asdict) or make a serializer.

**Comment by ferntheplant** [comment]
I can pickle a dataclass.

```python
import pickle
from dataclasses import dataclass

@dataclass
class Foo:
    foo: str

def main():
    x = Foo(foo="hello world")
    print(x) # Foo(foo='hello world')
    serialized = pickle.dumps(x)
    print(serialized) # binary data
    deserialized = pickle.loads(serialized)
    print(deserialized) # Foo(foo='hello world')
    print(x == deserialized) # true

if __name__ == '__main__':
    main()
```

**Comment by JonasKs** [comment]
Huh, weird. I am not home and won't be able to look into this for a week or so. Could you create a PR with a failing test (and a fix if you're able to find one)?

**Comment by ferntheplant** [comment]
Ah looks like my google-fu wasn't on point the other day when I filed this.

[This stack overflow answer](https://stackoverflow.com/a/50472787) explains exactly what happened in my toy case. `pickle` correctly serialized the module name to `__main__` even though the file is something like `workery.py`. When the arq worker runs however `__main__` is a different file altogether so pickle can't locate my dataclass (as seen in the error message `Can't get attribute 'FooParams' on <module '__main__'`).

I moved my dataclass definitions out from the file that I run to enqueue jobs and everything is working. Sorry to bother. Thanks!
